scipy/stats/_stats_py.py

```diff
@@ -5636,6 +5632,14 @@ def count_rank_tie(ranks):
     return res
 
 
+def _weightedtau_n_samples(kwargs):
+    rank = kwargs.get('rank', False)
+    return 2 if rank in {True, False} else 3
+
+
+@_axis_nan_policy_factory(_pack_CorrelationResult, default_axis=None, n_samples=2,
```

Comment:
Possible mistake?
```suggestion
@_axis_nan_policy_factory(_pack_CorrelationResult, default_axis=None, n_samples=_weightedtau_n_samples,
```

Reply:
Yup : ) probably got distracted by writing it. Should test it, too